DllImportGenerator/DllImportGenerator/TypePositionInfo.cs

```diff
@@ -82,9 +82,9 @@ private TypePositionInfo()
 
         public MarshallingInfo MarshallingAttributeInfo { get; init; }
 
-        public static TypePositionInfo CreateForParameter(IParameterSymbol paramSymbol, DefaultMarshallingInfo defaultInfo, Compilation compilation, GeneratorDiagnostics diagnostics)
+        public static TypePositionInfo CreateForParameter(IParameterSymbol paramSymbol, DefaultMarshallingInfo defaultInfo, Compilation compilation, GeneratorDiagnostics diagnostics, INamedTypeSymbol scopeSymbol)
```

Comment:
Ugh. These two functions (i.e. `CreateForParameter` and `CreateForType`) are hideous... Let's make a mental note that they should not be exposed as is when we enable extending the source generator.

/cc @elinor-fung